codex/contracts/market.nim

```diff
@@ -43,17 +43,20 @@ method getRequest(market: OnChainMarket,
     return none StorageRequest
 
 method getHost(market: OnChainMarket,
-               id: array[32, byte]): Future[?Address] {.async.} =
-  let address = await market.contract.getHost(id)
+               requestId: array[32, byte],
```

Comment:
Should we use `distinct` for this datatypes?

Reply:
Yes, that is a good idea. I'd like to address that in a separate PR, if that's ok.